### PR TITLE
Add dark mode support for images

### LINKED ISSUE #93: Add dark mode support to images
### Motivation
The migration to the new docfx modern template brings with it support for dark mode, but the existing images in this repository have not been updated and are unreadable in dark mode.

#### Screenshots 
- We could provide dark mode versions of them, but for now they are readable (if a bit glaring on dark mode).

#### Bonsai workflows
- They need to be converted to the new Bonsai workflow container in docfx-tools, but that is a separate issue/PR I will raise. They are readable for the time being.

#### Flowcharts
- Flowcharts can be converted into mermaid.js graphs which support dark mode, but I will say that positioning of nodes is tricky especially for non-linear/circular graphs. Even though I have tried the alternative graph options (block-beta, state diagram), the flowchart option remains the most customizable. Alternatively if the aesthetics of the new mermaid.js flowcharts are undesirable, editing or inverting the colors remains an option (outlined below).

#### SVGs
- The bulk of the images which need to be converted. The colors can be inverted (see discussion in alternative solutions) but I prefer picking certain colors to amend.

In some cases, this can be as simple as adding a style block near the top of the SVG to override inbuilt styles when dark mode is detected. The <code>!important</code> tag ensures that this takes precedence over inline styles. No other modifications to the SVG need to be added. 
``` js
<style>
    @media (prefers-color-scheme: dark) {
    path[style*="stroke:#000000"] {stroke: white !important;}
    path[style*="fill:#000000"] {fill: white !important;}   
    g[style*="fill:#000000"] {fill: white !important;}} 
 </style>
```
Unfortunately in most cases (the reactive operator marble diagrams for instance) the underlying SVG code isn't so clean, and necessitates additional modifications to the SVG to find and group together various paths by surrounding them with a unique <code>g id</code>. Its not hard to do (takes a few minutes per graph), just laborious. Then a style such as this can be applied.

``` js
<style>
    @media (prefers-color-scheme: dark) {
    path[stroke="#000"] {stroke: white !important;}
    path[fill="#000"] {fill: white !important;}
    path[fill="#fff"] {fill: #212529 !important;}
    g[id="text"] path, g[id="arrowheads"], path[id="g1-106"] {fill: white !important;}
    g[id="subscribe-text"] {fill: #eb008a !important;}
    g[id="subscribe-arrows"] path {stroke: #eb008a !important;}}
</style>
```
In the end I chose to do these changes:
- Keep original marble colors
- Convert black text and outlines to white
- Convert white fills to the background color of docfx dark mode theme
- Convert blue subscribe-text and subscribe-arrows to pink.

#### Invert filter  
``` js
 <style>
    @media (prefers-color-scheme: dark) { svg {filter: invert(1) } }
 </style>
```
This was the first option that I tried which inverts the colors. This is easier and faster to apply but it has several drawbacks: 
1. Colors are a bit ugly but this is subjective of course.
2. In some instances, original colors are important (for instance, colors for the operator nodes). 
3. Some containers have a white fill instead of a blank fill, which went inverted becomes black, which stands out against the dark mode background (which is more of a dark blueish/grey background). This was present in like 30-50% of the images. In most cases this is more of an aesthetic thing, but in some cases it is functionally important. For instance, some marbles which look blank in white mode are supposed to represent that the reactive operator doesn't care about some attribute in the observable sequence. When they become black on dark grey in dark mode however they look like they have some other function or transformation. 
 
For consistency's sake, I chose to go with the solution outlined above.

#### External stylesheet
Instead of editing all the SVGs I considered using an external CSS style, but with the variability of all the different SVGs, and the possibility that it might affect future SVGs in unexpected ways, I decided against it.

### PR DESCRIPTION
**Pull request by banchan86**
Fixes #93 

Online preview of this PR and #97 https://banchan86.github.io/bonsai_docs_test/

### CODE REVIEW

**Review by glopesdev — COMMENTED** [review]
PR looks great, must have been a lot of work @banchan86. Added minor comments below, really looking forward to finally merging this.

File: images/language-temperature-effects.svg
[the remark this answers is not in the document]
Comment (banchan86) [reply images/language-temperature-effects.svg:1]: good catch! I implemented the same solution that I used for the other images.

File: images/language-marblediagram-error.svg [hunk 1]
```diff
@@ -482,7 +488,7 @@
     <path
        id="use57"
        d="m 141.17228,-6.9703421 -1.61162,-1.6035595 c -0.0967,-0.096697 -0.11281,-0.1128131 -0.17728,-0.1128131 -0.0806,0 -0.16116,0.072523 -0.16116,0.161162 0,0.056406 0.0161,0.072522 0.10476,0.1611615 l 1.61161,1.6196757 -1.61161,1.6196755 c -0.0887,0.088639 -0.10476,0.1047551 -0.10476,0.1611616 0,0.088639 0.0806,0.1611616 0.16116,0.1611616 0.0645,0 0.0806,-0.01612 0.17728,-0.1128131 l 1.60356,-1.6035593 1.66802,1.6680244 c 0.0161,0.00805 0.0725,0.04835 0.12087,0.04835 0.0967,0 0.16116,-0.072522 0.16116,-0.1611616 0,-0.01612 0,-0.04835 -0.0242,-0.088639 -0.008,-0.01612 -1.28929,-1.281236 -1.6922,-1.692198 l 1.47463,-1.4746303 c 0.0403,-0.048348 0.16116,-0.1531036 0.20145,-0.2014521 0.008,-0.016118 0.0403,-0.048349 0.0403,-0.1047548 0,-0.088639 -0.0645,-0.161162 -0.16116,-0.161162 -0.0645,0 -0.0967,0.032232 -0.18534,0.1208713 z"
-       style="stroke-width:0.389986" />
+       style="stroke-width:0.389986;stroke:#000000"/>
```
[the remark this answers is not in the document]
Comment (banchan86) [reply images/language-marblediagram-error.svg]: I changed the stroke definition to a fill instead so it doesn't change the width

File: images/reactive-first.svg [hunk 2]
```diff
@@ -1,14 +1,23 @@
 <?xml version='1.0' encoding='UTF-8'?>
 <!-- This file was generated by dvisvgm 2.11.1 -->
 <svg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' width='425.192834pt' height='201.035811pt' viewBox='-146.287854 -153.360224 425.192834 201.035811'>
+<style>
+    @media (prefers-color-scheme: dark) {
+    path[stroke="#000"] {stroke: #eee !important;}
+    path[fill="#000"] {fill: #eee !important;}
+    path[fill="#fff"] {fill: #212529 !important;}
```
[the remark this answers is not in the document]
Comment (banchan86) [reply images/reactive-first.svg:8]: gotcha, yea I also noticed a mix of both transparent and filled boxes throughout the SVGs but I only filled in the ones that had a visual impact.

File: tutorials/state-machines.md
[the remark this answers is not in the document]
Comment (banchan86) [reply tutorials/state-machines.md:1]: I'm happy with these changes, they look cleaner both visually and in the code (especially for the first one), so I copied the state diagrams over.

There was an issue that cropped up though (https://banchan86.github.io/bonsai_docs_test/tutorials/state-machines.html), the transition text renders differently (the boundary box is semi-transparent) compared to GitHub or the official mermaid.js website and is obstructed.

Although the local docfx installation is updated to the latest release its still using an old version (v10.9.1) of mermaid.js.    It still doesn't too bad, and will probably be updated in the future, if you are ok with how it looks we can use the state diagrams.

**Review by glopesdev — APPROVED** [review]

File: tutorials/state-machines.md
[the remark this answers is not in the document]
Comment (PathogenDavid) [reply tutorials/state-machines.md:1]: Gonçalo nerd sniped me on this last week but I only just now got a chance to look. I think it looks fine, but I do love a good mystery.

The edge label background was indeed [added in Mermaid 11](https://github.com/mermaid-js/mermaid/commit/1e864a508d7f030b79e9bf58667354057694b1fc#diff-ac12e8d6d73c6cb094fc8bc5e418a17157269774f16cc891532b947918ee88f1R90).

Docfx [recently updated Mermaid from 10 to 11](https://github.com/dotnet/docfx/commit/02a052ba0b7515af08c16aae401fa793af38ac46), but that change hasn't landed in a stable release yet.

Probably easiest to just wait for the docfx update, but if you wanna be impatient you can add this quick and dirty CSS to the dark theme:

```css
.mermaid svg .edgeLabel { background-color: hsl(0, 0%, 34%); }
```

Looks like this:

Screenshot of diagram with a background behind the state transition labels

**Pull request merged**